From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct fib_nh_exception
Patch-mainline: never, kabi

In 3.12.56, commit 27b6d267f4d3652dee80ced90800904ed0df6a05 (route:
check and remove route cache when we get route), upstream commit
deed49df7390d5239024199e249190328f1651e7 added rcu into struct
fib_nh_exception. It indeed changed the layout of the structure.

But given this is only ipv4-routing-internal structure, protect the
member the usual way, by #ifdef __GENKSYMS__.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/net/ip_fib.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/include/net/ip_fib.h
+++ b/include/net/ip_fib.h
@@ -59,7 +59,9 @@ struct fib_nh_exception {
 	struct rtable __rcu		*fnhe_rth_input;
 	struct rtable __rcu		*fnhe_rth_output;
 	unsigned long			fnhe_stamp;
+#ifndef __GENKSYMS__
 	struct rcu_head			rcu;
+#endif
 };
 
 struct fnhe_hash_bucket {
